lib/src/clients/api_client.dart

```diff
@@ -3,21 +3,30 @@ part of uhst_clients;
 /// [ApiClient] Wraps [RelayClient] and is responsible for getting a
 /// relayUrl from the UHST public relays directory.
 class ApiClient implements UhstRelayClient {
-  ApiClient() : relayUrlsProvider = RelayUrlsProvider();
+  ApiClient()
+      : relayUrlsProvider = const RelayUrlsProvider(
+          networkClient: NetworkClient(),
+        );
   RelayUrlsProvider relayUrlsProvider;
   late RelayClient relayClient;
 
   @override
   Future<ClientConfiguration> initClient({required String hostId}) async {
     final relayUrl = await relayUrlsProvider.getBestRelayUrl(hostId);
-    relayClient = RelayClient(relayUrl: relayUrl);
+    relayClient = RelayClient(
+      relayUrl: relayUrl,
+      networkClient: const NetworkClient(),
```

Comment:
why we pass default NetworkClient in constructor? If it will be default should be instantiated by RelayClient .

Reply:
It is the moment of optimization - if you can make class @immutable then in my option it is needed to do it- it will save time and resources next time, because of dart nature. As an example, you can read about const differences [this article](https://news.dartlang.org/2012/06/const-static-final-oh-my.html)

Reply:
It is internal to the RelayClient implementation and I don't think it's relevant to the RelayClient API.

Reply:
also this const is not reused, the reference to it exists only inside of the instance of RelayClient. If it was something like getNetworkClient() which always returns the same const maybe it would make sense, but passing a new "const" instance in the contructor?